From fc301b35b175ef24b7f6d0946c2df1bdd269dc77 Mon Sep 17 00:00:00 2001
From: Fredy P <fredy.pulido@savoirfairelinux.com>
Date: Mon, 09 Dec 2019 10:53:14 -0500
Subject: [PATCH] ffmpeg: linux nvenc,nvdec,cuvid for x86 only

Removes nvenc, nvdec and cuvid from linux desktop compilation
Enables nvenc, nvdec and cuvid for i386 and x86_64
Disables cuvid for arm 32 and 64 bits.

Source: https://review.jami.net/c/ring-daemon/+/13282

Change-Id: If5406611602dba23e02ebce281aeaeb43bc4d53c
---

diff --git a/contrib/src/ffmpeg/rules.mak b/contrib/src/ffmpeg/rules.mak
index 32e32b1..649ab2d 100644
--- a/daemon/contrib/src/ffmpeg/rules.mak
+++ b/daemon/contrib/src/ffmpeg/rules.mak
@@ -199,11 +199,33 @@
 	--enable-decoder=h264_mediacodec \
 	--enable-decoder=mpeg4_mediacodec
 # ASM not working on Android x86 https://trac.ffmpeg.org/ticket/4928
+# ffnvcodec is not supported on ARM then we enable it here for i386 and x86
+# FIXME we have same for i386 and X86_64 does a merge makes sense?
 ifeq ($(ARCH),i386)
-FFMPEGCONF += --disable-asm
+FFMPEGCONF += --disable-asm \
+	      --enable-cuvid \
+	      --enable-ffnvcodec \
+	      --enabl--enable-nvdec \
+	      --enabl--enable-nvenc \
+	      --enabl--enable-hwaccel=h264_nvdec \
+	      --enabl--enable-hwaccel=hevc_nvdec \
+	      --enabl--enable-hwaccel=vp8_nvdec \
+	      --enabl--enable-hwaccel=mjpeg_nvdec \
+	      --enabl--enable-encoder=h264_nvenc \
+	      --enabl--enable-encoder=hevc_nvenc
 endif
 ifeq ($(ARCH),x86_64)
-FFMPEGCONF += --disable-asm
+FFMPEGCONF += --disable-asm \
+	      --enable-cuvid \
+	      --enable-ffnvcodec \
+	      --enabl--enable-nvdec \
+	      --enabl--enable-nvenc \
+	      --enabl--enable-hwaccel=h264_nvdec \
+	      --enabl--enable-hwaccel=hevc_nvdec \
+	      --enabl--enable-hwaccel=vp8_nvdec \
+	      --enabl--enable-hwaccel=mjpeg_nvdec \
+	      --enabl--enable-encoder=h264_nvenc \
+	      --enabl--enable-encoder=hevc_nvenc
 endif
 else
 # Desktop Linux
@@ -223,17 +245,7 @@
 	--enable-hwaccel=mjpeg_vaapi \
 	--enable-encoder=h264_vaapi \
 	--enable-encoder=vp8_vaapi \
-	--enable-encoder=mjpeg_vaapi \
-	--enable-cuvid \
-	--enable-ffnvcodec \
-	--enable-nvdec \
-	--enable-nvenc \
-	--enable-hwaccel=h264_nvdec \
-	--enable-hwaccel=hevc_nvdec \
-	--enable-hwaccel=vp8_nvdec \
-	--enable-hwaccel=mjpeg_nvdec \
-	--enable-encoder=h264_nvenc \
-	--enable-encoder=hevc_nvenc
+	--enable-encoder=mjpeg_vaapi
 endif
 endif
 
@@ -279,7 +291,8 @@
 
 # ARM stuff
 ifeq ($(ARCH),arm)
-FFMPEGCONF += --arch=arm
+FFMPEGCONF += --arch=arm \
+	      --disable-cuvid
 ifdef HAVE_ARMV7A
 FFMPEGCONF += --cpu=cortex-a8
 endif
@@ -290,10 +303,12 @@
 
 # ARM64 stuff
 ifeq ($(ARCH),aarch64)
-FFMPEGCONF += --arch=aarch64
+FFMPEGCONF += --arch=aarch64 \
+	      --disable-cuvid
 endif
 ifeq ($(ARCH),arm64)
-FFMPEGCONF += --arch=aarch64
+FFMPEGCONF += --arch=aarch64 \
+	      --disable-cuvid
 endif
 
 # Windows
